docs/ecosystem-roadmap.md

```diff
@@ -0,0 +1,101 @@
+
+## Definitions
+
+It is highly recommended to familiarize yourself with the [definitions](definitions.md) behind common terminology in this space before proceeding with the remainder of this document.
+
+## Architecture
+
+The general architecture of the system is in four parts: client, device, server, and terminal. Each is connected to the adjacent items in the list with the two ends as the respective source and destination to command-line communication.
+
+### Client
+
+A client is a command-line application that interprets a user's desires through a text-based interface, performs work or dispatches further commands, and returns a text representation of the result. On Windows, the console API or the standard console handles with device control APIs provide a communications layer between the client and the device.
+
+### Device
+
+The device is an intermediate message handling communications layer between two processes, the client and the server. On Windows, this is the console driver, while on other platforms, it is the TTY or PTY device. Other devices like files, pipes, and sockets may be used as this communication channel if the entire transaction is in plain text or contains [virtual terminal sequences](console-virtual-terminal-sequences.md), but not with [Windows Console APIs](console-functions.md).
+
+### Server
+
+The server interprets the requested API calls or messages from the client. On Windows in the classic operating mode, it also creates a user interface to present the output to the screen and collects input to send back in response messages to the client via the driver, like a terminal bundled in the same module. With the [pseudoconsole](pseudoconsoles.md) mode, it instead is only a translator to present this information in [virtual terminal sequences](console-virtual-terminal-sequences.md) to an attached Terminal.
+
+### Terminal
+
+The terminal is the final layer providing graphical display and interactivity services to the user. It is responsible for capturing input and encoding it as [virtual terminal sequences](console-virtual-terminal-sequences.md) to eventually reach the client's `STDIN` and decoding and presenting the plain text and _virtual terminal sequences_ it received back from the client's `STDOUT`.
+
+#### Further Connections
+
+As an addendum, further connections can be performed by chaining applications that serve multiple roles into one of the endpoints. For instance, an SSH session has two roles: it is a Terminal for the command-line application running on one device, but it forwards all received information on to a Client role on another device. This chaining can occur indefinitely across devices and contexts offering broad scenario flexibility.
+
+On non-Windows platforms, the Server and Terminal roles are a single unit because there is no need for a translation compatibility layer between an API set and [virtual terminal sequences](console-virtual-terminal-sequences.md) as there is on Windows.

```

Comment:
```suggestion
On non-Windows platforms, the server and terminal roles are a single unit because there is no need for a translation compatibility layer between an API set and [virtual terminal sequences](console-virtual-terminal-sequences.md) as there is on Windows.

```